argo_cd/tests/test_argo_cd.py

```diff
@@ -0,0 +1,18 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+
+import pytest
+
+from datadog_checks.argo_cd import ArgoCdCheck
+
+CHECK_NAME = 'argo_cd'
+
+
+@pytest.mark.integration
+def test_check_all_metrics(aggregator, dd_run_check, mock_argo_cd):
+    instance = {'openmetrics_endpoint': 'http://localhost:8082/metrics'}
+    c = ArgoCdCheck(CHECK_NAME, {}, [instance])
+    dd_run_check(c)
+    aggregator.assert_metric("argocd.app_info", count=4, value=1)
```

Comment:
Why do you only check 1 metric?

Reply:
When checking every metrics, I have this error, do you know why ? 

```
E       AssertionError: Needed at least 1 candidates for 'argocd.app_k8s_request_total', got 0
E         Expected:
E                 MetricStub(name='argocd.app_k8s_request_total', type=None, value=None, tags=None, hostname=None, device=None)
E         Similar submitted:
E         Score   Most similar
E         0.56    MetricStub(name='argocd.app_info', type=0, value=1.0, tags=['dest_namespace:namespace4', 'dest_server:https://kubernetes.default.svc', 'endpoint:http://localhost:50000/metrics', 'health_status:Healthy', 'name:appname4', 'namespace:argocd', 'operation:', 'project:default', 'repo:git@github.com/myorg/repo-gitops', 'sync_status:Synced'], hostname='stubbed.hostname', device=None)
E         0.56    MetricStub(name='argocd.app_info', type=0, value=1.0, tags=['dest_namespace:namespace3', 'dest_server:https://kubernetes.default.svc', 'endpoint:http://localhost:50000/metrics', 'health_status:Healthy', 'name:appname3', 'namespace:argocd', 'operation:', 'project:default', 'repo:git@github.com/myorg/repo-gitops', 'sync_status:Synced'], hostname='stubbed.hostname', device=None)
E         0.56    MetricStub(name='argocd.app_info', type=0, value=1.0, tags=['dest_namespace:namespace2', 'dest_server:https://kubernetes.default.svc', 'endpoint:http://localhost:50000/metrics', 'health_status:Healthy', 'name:appname2', 'namespace:argocd', 'operation:', 'project:default', 'repo:git@github.com/myorg/repo-gitops', 'sync_status:Synced'], hostname='stubbed.hostname', device=None)
E         0.56    MetricStub(name='argocd.app_info', type=0, value=1.0, tags=['dest_namespace:namespace1', 'dest_server:https://kubernetes.default.svc', 'endpoint:http://localhost:50000/metrics', 'health_status:Healthy', 'name:appname1', 'namespace:argocd', 'operation:', 'project:default', 'repo:git@github.com/myorg/repo-gitops', 'sync_status:Synced'], hostname='stubbed.hostname', device=None)
E         0.54    MetricStub(name='argocd.cluster_api_resources', type=0, value=89.0, tags=['endpoint:http://localhost:50000/metrics', 'server:https://kubernetes.default.svc'], hostname='stubbed.hostname', device=None)
E         0.51    MetricStub(name='argocd.cluster_info', type=0, value=1.0, tags=['endpoint:http://localhost:50000/metrics', 'k8s_version:1.21+', 'server:https://kubernetes.default.svc'], hostname='stubbed.hostname', device=None)
E         0.48    MetricStub(name='argocd.cluster_api_resource_objects', type=0, value=1077.0, tags=['endpoint:http://localhost:50000/metrics', 'server:https://kubernetes.default.svc'], hostname='stubbed.hostname', device=None)
E         0.40    MetricStub(name='argocd.cluster_cache_age_seconds', type=0, value=16003.0, tags=['endpoint:http://localhost:50000/metrics', 'server:https://kubernetes.default.svc'], hostname='stubbed.hostname', device=None)
E         0.33    MetricStub(name='argocd.kubectl_exec_pending', type=0, value=0.0, tags=['command:apply', 'endpoint:http://localhost:50000/metrics', 'hostname:argocd-application-controller-76b4484b8f-wwr47'], hostname='stubbed.hostname', device=None)
E       assert False
```

Reply:
Because maybe that particular metric is not submitted during the test run, try to assert as many metrics as you can get reliably during the test execution. I'm not familiar with argo but in general in many systems there are metrics that are only emitted under certain conditions